lib/puppet/resource_api/transport.rb

```diff
@@ -35,6 +35,12 @@ def connect(name, connection_info)
   end
   module_function :connect # rubocop:disable Style/AccessModifierDeclarations
 
+  def inject_device(name, transport)
+    transport_wrapper = Puppet::ResourceApi::Transport::Wrapper.new(name, transport)
+    Puppet::Util::NetworkDevice.instance_variable_set(:@current, transport_wrapper)
```

Comment:
I updated bolt to prefer `set_device` once that gets into puppet but I guess I forgot to push it up.

```
    if  Puppet::Util::NetworkDevice.respond_to?(:set_device)
      Puppet::Util::NetworkDevice.set_device(conn_info['type'], transport_wrapper)
    else
      Puppet::Util::NetworkDevice.instance_variable_set(:@current, transport_wrapper)
    end
```

Reply:
Yeah, this is exactly to code @DavidS suggested earlier today :)